### PR TITLE
ZK-3958: Checkbox add new molds (switch, toggle)

### PR DESCRIPTION
**Pull request by charlesqqq**

### CODE REVIEW

File: zul/src/archive/web/js/zul/wgt/mold/checkbox.js [hunk 1]
```diff
@@ -15,9 +15,11 @@ it will be useful, but WITHOUT ANY WARRANTY.
...
 	//tabindex attribute will be set in input element
-	out.push('<span', this.domAttrs_({tabindex: 1}), '><input type="checkbox" id="', uuid,
-			'-real"', this.contentAttrs_(), '/><label for="', uuid,
-			'-real" id="', uuid, '-cnt"', this.domTextStyleAttr_(), 
-			' class="', this.$s('content') ,'">', this.domContent_(),
-			'</label></span>');
+	out.push('<span', this.domAttrs_({tabindex: 1}), '>');
+	out.push('<input type="checkbox" id="', uuid, '-real"', this.contentAttrs_(), '/>');
+	if (zk.feature.ee)
+		out.push('<label for="', uuid, '-real" id="', uuid, '-mold" class="', this.$s(this.getMold()), '"/>');
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/wgt/mold/checkbox.js]: Will have two `<label>` on ZK EE?

Comment (charlesqqq) [reply zul/src/archive/web/js/zul/wgt/mold/checkbox.js]: one for mold, another for text

File: zktest/src/archive/test2/F86-ZK-3958.zul [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	<vlayout>
+		<checkbox id="checkbox" label="Checkbox"/>
+		<hlayout>
+			<button label='Default' onClick='checkbox.setMold("default")'/>
+			<button label='Switch' onClick='checkbox.setMold("switch")'/>
+			<button label='Toggle' onClick='checkbox.setMold("toggle")'/>
+		</hlayout>
+		<hlayout>
+			<button label="isChecked" onClick="Clients.log(checkbox.isChecked().toString())"/>
+			<button label="setChecked" onClick="checkbox.setChecked(!checkbox.isChecked())"/>
```
Comment (scribetw) [inline zktest/src/archive/test2/F86-ZK-3958.zul]: Please add setDisabled for style checking

File: zktest/test/java/org/zkoss/zktest/zats/test2/F86_ZK_3958Test.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		
+		verifyClassChange(jq(".z-button:contains(Toggle)"), jq(toggleClass + onClass));
+		
+		verifyClassChange(moldLocation, jq(toggleClass + offClass));
+		
+		verifyChecked("false");
+		
+		click(jq(".z-button:contains(setDisabled)"));
+		waitResponse();
+		Assert.assertEquals("0.5", moldLocation.css("opacity"));
```
Comment (scribetw) [inline zktest/test/java/org/zkoss/zktest/zats/test2/F86_ZK_3958Test.java]: Not good for customizing.

Preferred: `setDisabled` toggleClass('z-checkbox-disabled'). You check if the class is applied.

**Pull request merged**